File: solutions/M. I am 신뢰예요.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int n, m;
vector<int> cats;
vector<int> students;

int compare(int &a, int &b) {
    return a < b;
}

void solve1() {
    for (int index = 0; index < n; index++) { // n
        int cat_height = cats[index];
        int cnt = upper_bound(students.begin(), students.end(), cat_height) - students.begin(); // logm
        cout << cnt << ' ';
    }
    cout << endl;
}

int upperBound(int target, vector<int> &Array) {
    int left = 0, right = Array.size() - 1;

    while (left <= right) {//
        int mid = (left + right) / 2;
        if (Array[mid] > target) {
            right = mid - 1;
        } else {
            left = mid + 1;
        }
    }

    return left;
}


void solve2() {
    // 모든 고양이들의 키에 대해서 '착한인간구별공식' 을 만족하는 학생의 마지막 인덱스 파악
    for (int index = 0; index < n; index++) {
        int cat_height = cats[index];
        int count = upperBound(cat_height, students);
        cout << count << ' ';
    }
    cout << endl;
}

int main() {
    ios::sync_with_stdio(false), cin.tie(nullptr);
    cin >> n >> m;
    cats.resize(n);
    students.resize(m);

    for (int index = 0; index < n; index++) {
        cin >> cats[index];
    }

    for (int index = 0; index < m; index++) {
        cin >> students[index];
    }

    sort(students.begin(), students.end(), compare); // 학생키 오름차순 정렬

    solve1();

    return 0;
}

File: solutions/ch/M.I am 신뢰예요.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Scanner;

// M. I am 신뢰예요
public class Main {

    static int M;
    static int UpperBound(List<Integer> students, int target) {
        int lt = 0;
        int rt = M;
        while (lt < rt) {
            int mid = lt + (rt - lt) / 2;

            if (students.get(mid) <= target) {
                lt = mid + 1;
            } else {
                rt = mid;
            }
        }
        return lt;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        M = sc.nextInt();
        ArrayList<Integer> cats = new ArrayList<>();
        ArrayList<Integer> students = new ArrayList<>();
        for(int i = 0 ; i < N ; i ++){
            cats.add(sc.nextInt());
        }

        for(int i = 0 ; i < M ; i ++){
            students.add(sc.nextInt());
        }

        Collections.sort(students);
        String answer = "";
        for(int i = 0 ; i < N ; i ++) {
            answer += UpperBound(students, cats.get(i)) + " ";
        }
        System.out.println(answer);
    }
}
